ui/components/component-library/button-primary/button-primary.stories.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import { ALIGN_ITEMS, DISPLAY } from '../../../helpers/constants/design-system';
+import Box from '../../ui/box/box';
+import { ICON_NAMES } from '../icon';
+import { ButtonPrimary } from './button-primary';
+import { BUTTON_PRIMARY_SIZES } from './button-primary.constants';
+import README from './README.mdx';
+
+const marginSizeControlOptions = [
+  undefined,
+  0,
+  1,
+  2,
+  3,
+  4,
+  5,
+  6,
+  7,
+  8,
+  9,
+  10,
+  11,
+  12,
+  'auto',
+];
+
+export default {
+  title: 'Components/ComponentLibrary/ButtonPrimary',
+  id: __filename,
+  component: ButtonPrimary,
+  parameters: {
+    docs: {
+      page: README,
+    },
+  },
+  argTypes: {
+    as: {
+      control: 'select',
+      options: ['button', 'a'],
+      table: { category: 'button base props' },
+    },
+    block: {
+      control: 'boolean',
+      table: { category: 'button base props' },
+    },
+    children: {
+      control: 'text',
+    },
+    className: {
+      control: 'text',
+    },
+    danger: {
+      control: 'boolean',
+    },
+    disabled: {
+      control: 'boolean',
```

Comment:
the props for ButtonBase don't seem to be transferring over. I think it would be a good idea to add a test or so to ensure the ButtonBase props are working